examples/js-sub.c

```diff
@@ -130,21 +171,49 @@ int main(int argc, char **argv)
 
     if (s == NATS_OK)
     {
-        if (pull)
+        if (pull && async)
+        {
+            jsOpts.PullSubscribeAsync.MaxMessages = (int) total;
+
+            // Defalut values, change as needed.
+            jsOpts.PullSubscribeAsync.FetchSize = 128;  // ask for 128 messages at a time
+            jsOpts.PullSubscribeAsync.NoWait = false;
+            jsOpts.PullSubscribeAsync.Timeout = 0;      // for the entire subscription, in milliseconds
```

Comment:
This is fine,
But I think that each pull request should have 30s expires default value.
Otherwise if server looses the Pull Requst, client will never know about it.

Similarly, HeartBeats should be enabled by default, with a value of 15 seconds (half of expires, which is maximum).

Reply:
@Jarema First, 1/5 I don't think that subscription-level Expires/Timeout should be defaulted on. By default, a sub should live until Unsubscribe, that's the idiom. A separate FetchTimeout (was Batch a better name?) would solve this issue but also add complexity and potential confusion.

Second, if the heartbeats are ON, and the server loses a pull request, the client _will_ know about because of the missing heartbeat error. So, if we make heartbeat ON by default at some number of seconds why add a default timeout?

Reply:
I agree on not having timeout on subscription. In fact, none other clients have that option at all.
If you want to stop subscription, stop it explicitly, or do drain. We never do timeouts like thoge.

What I meant however, is `expires` value for each `pull requst`, and `heartbeats` for those pull requests too.